shapes/README.md

```diff
@@ -0,0 +1,84 @@
+# Shapes
```

Comment:
There seems to be a large number of revisions needed in the README based on the updates to the various ShEx shapes. Instead of rewriting in the PR, probably worth reworking and resubmitting, even in a separate PR after the ShEx shapes are approved.

Reply:
I applied most of the changes suggested but I agree, we should revise the whole README once we have the final shapes.

Reply:
I did some changes, but now waiting for @timbl final review, then we'll see what needs to be changed further.